digitaltwin_client/samples/digitaltwin_sample_device/digitaltwin_sample_device.c

```diff
@@ -78,73 +78,17 @@ static const int digitalTwinSampleDevice_sendTelemetryFrequency = 20;
 // END TODO section
 //
 
-// State of DigitalTwin registration process.  We cannot proceed with DigitalTwin until we get into the state APP_DIGITALTWIN_REGISTRATION_SUCCEEDED.
-typedef enum APP_DIGITALTWIN_REGISTRATION_STATUS_TAG
-{
-    APP_DIGITALTWIN_REGISTRATION_PENDING,
-    APP_DIGITALTWIN_REGISTRATION_SUCCEEDED,
-    APP_DIGITALTWIN_REGISTRATION_FAILED
-} APP_DIGITALTWIN_REGISTRATION_STATUS;
-
-//
-// DigitalTwinSampleDevice_InterfacesRegistered is invoked when the interfaces have been registered or failed.
-// The userContextCallback pointer is set to whether we succeeded or failed and checked by thread blocking
-// for registration to complete.
-static void DigitalTwinSampleDevice_InterfacesRegistered(DIGITALTWIN_CLIENT_RESULT dtInterfaceStatus, void *userContextCallback)
-{
-    APP_DIGITALTWIN_REGISTRATION_STATUS* appDigitalTwinRegistrationStatus = (APP_DIGITALTWIN_REGISTRATION_STATUS*)userContextCallback;
-
-    if (dtInterfaceStatus == DIGITALTWIN_CLIENT_OK)
-    {
-        LogInfo("Interface registration callback invoked, interfaces have been successfully registered");
-        *appDigitalTwinRegistrationStatus = APP_DIGITALTWIN_REGISTRATION_SUCCEEDED;
-    }
-    else
-    {
-        LogError("Interface registration callback invoked with an error=<%s>", MU_ENUM_TO_STRING(DIGITALTWIN_CLIENT_RESULT,dtInterfaceStatus));
-        *appDigitalTwinRegistrationStatus = APP_DIGITALTWIN_REGISTRATION_FAILED;
-    }
-}
-
-APP_DIGITALTWIN_REGISTRATION_STATUS appDigitalTwinRegistrationStatus = APP_DIGITALTWIN_REGISTRATION_PENDING;
-
 // Invokes DigitalTwin_DeviceClient_RegisterInterfacesAsync, which indicates to Azure IoT which DigitalTwin interfaces this device supports.
 // The DigitalTwin Handle *is not valid* until this operation has completed (as indicated by the callback DigitalTwinSampleDevice_RegisterDigitalTwinInterfacesAndWait being invoked).
 static DIGITALTWIN_CLIENT_RESULT DigitalTwinSampleDevice_RegisterDigitalTwinInterfacesAndWait(DIGITALTWIN_DEVICE_CLIENT_HANDLE dtDeviceClientHandle, DIGITALTWIN_INTERFACE_CLIENT_HANDLE* interfaceClientHandles, int numInterfaceClientHandles)
```

Comment:
>DigitalTwinSampleDevice_RegisterDigitalTwinInterfacesAndWait [](start = 33, length = 60)

Let's remove this helper function since AndWait() is N/A now.  Can just move the DigitalTwin_DeviceClient_RegisterInterfacesAsync back into the main().

Reply:
Same feedback with LL version...

---
In reply to: [387337016](https://github.com/Azure/azure-iot-sdk-c/pull/1439#discussion_r387337016) [](ancestors = 387337016)